backend/db/utils.py

```diff
@@ -167,6 +166,16 @@ def update_db_status_var(key: str, val: str, local=False):
         sql_str = f"INSERT INTO public.manage (key, value) VALUES (:key, :val);"
         run_sql(con, sql_str, {'key': key, 'val': val})
 
+def check_db_status_var(key: str,  local=False):
+    """Check the value of a given variable the `manage`table """
+    with get_db_connection(schema='', local=local) as con2:
+        results: List = sql_query_single_col(con2, f"SELECT value FROM public.manage WHERE key = '{key}';")
+        return results[0] if results else None
```

Comment:
@hugwuoke This one would have been impossible to catch without doing a test run, which is my job for this PR. I think there were some recent refactoring changes. I think `sql_query_single_col()` is a new column, and it returns a list, so I had to grab the 0th item.

Reply:
I think some more clarification on how you tested might be helpful, I thought I tested things ok, I opened up the page locally and hit refresh and everything seemed fine. I think it might be helpful to get more information on how you tested so that I would be able to do more of that by myself in the future.

Reply:
Hmm, you know what. It might be possible that it was working correctly when you were programming this. I did a `git rebase develop` on this branch, and I don't remember if I encountered this before the rebase, or afterwards.

To clarify, basically I needed to add `results[0]`. I believe this came up towards the beginning of the script think it was when it was checking if the status was `'active'`, but what was coming back from this function call was `['active']`.

So now that I think of it, this may well have come up before I rebased. But since it didn't produce an error message, that would be why it would have been difficult for you to catch.

I think I noticed this because I clicked the button once, and then I clicked it a second time. And I wanted to check that the second click was behaving correctly. Since it wasn't able to correctly determine that the refresh was in the `'active'` state, I noticed that the second click resulted in it not exiting out as expected (e.g. I didn't see the `print` statement about it exiting out). 

---

In any case, this was a bit tricky to test out. I am not 100% confident in my testing results. I tried this a few ways ways. The first way, I set a breakpoint in the middle of the refresh, clicked the refresh button, let it sit there paused at the breakpoint, and then I hit the refresh button again. Another way I tested it, was rather than setting a breakpoint, I did `time.sleep()` a bunch of times. And hit the refresh button a 2nd time while those sleeps were running.

FYI as a safety measure I also temporarily commented some lines of code so that it was not actually going to run the refresh for real and  I didn't want it to update some of the DB variables.

But yeah, I want to test this out when we deploy this to dev. I'm not 100% confident in my results because some of the times that I ran this, I got the result that I expected. I saw the 2nd refresh started and exited out as expected. But other times, 2nd refresh started sequentially only after the 1st had completed. It's a bit tricky to test this kind of thing because it relies on the multithreading nature of the FastAPI web server to be working correctly. And I believe that it functions differently in the development environment than when deployed.